R/calc_coherence.R

```diff
@@ -0,0 +1,259 @@
+
+
+#TODO DOC: short explanation of implemented coherence measures
+
+#' Calculation of various coherence measures for topic models
+#'
+#'
+#'
+#' @param top_term_matrix A character matrix with the top words (entries = words) ranked starting with row 1 per topic (columnnames = topics). e.g.,
+#'                        For example, the output of text2vec::get_top_words(...).
+#' @param tcm A term co-occurrence matrix that serves as reference to calculate coherence scores for the terms in \code{top_term_matrix}.
+#' @param ndocs_tcm The number of documents that served to create the \code{tcm} to calculate probabilities from counts in the\code{tcm} used as input to some coherence measures.
+#' @param log_smooth_constant Smoothing constant to avoid logarithm of zero in calcualtions, for example, \code{log(log_smooth_constant + 0)}.
+#'                            By default \code{.1e-12} as used by Röder et al. (palmetto).
+#'                            Use \code{1} for UMass logratio as used by Mimno.
+#'                            Use \code{.01} for logratio as used in stm package.
+#' @param average_over_topics By default \code{FALSE} to return coherence score for each indivual topic.
+#'                            When set to \code{TRUE} the mean score over all topics is returned.
+#'
+#' @return A \code{data.table} showing the various coherence scores per topic (or average over all topics).
+#' @export
+#'
+#' @examples
```

Comment:
Would be great to provide example based on `movie_review` dataset
- create_dtm
- build lda and get top terms
- show how to calculate tcm (`crossprod(sign(dtm))`)

Reply:
Added an initial example based on `movie_review` which will have to be updated after further changes of the `coherence` function. In the example we can already see that several measures are not really informative (e.g. pmi) as intrinsic measures when using the full document.

Done via 45deee1